include/shpxx/shpfile.hpp

```diff
@@ -68,6 +105,35 @@ class shpfile_t
     //! z and m should be 0 when they are unused, but this is not guaranteed.
     [[nodiscard]] geometry::point_xyzm_t max_bound() const noexcept;
 
+    //! @brief Create an read-only iterator pointing to the beginning of the
+    //! file
+    //!
+    //! @tparam FeatureT The feature type that all data in the file should be
+    //! converted to. This must be compatible with the shape_type returned by
+    //! type()
+    //!
+    //! @throw incompatible_feature_type_error_t if FeatureT is not compatible
+    //! with the type of features indicated by the file header
+    //!
+    //! @return A new feature iterator which can be used to access the features
+    //! of the file as a specific feature type
+    template<shpxx::concepts::IsFeature FeatureT>
+    [[nodiscard]] const_feature_iterator_t<FeatureT> begin() const;
```

Comment:
Should this be `cbegin()` and `cend()`?

Reply:
Should probably have aliases for those. afaik, convention is for `cbegin() const` and `begin() const` to be identical and/or defer one to the other

Reply:
Added